source/common/filter/ext_authz.cc

```diff
@@ -0,0 +1,102 @@
+#include "common/filter/ext_authz.h"
+
+#include <cstdint>
+#include <string>
+
+#include "common/common/assert.h"
+#include "common/tracing/http_tracer_impl.h"
+
+#include "fmt/format.h"
+
+namespace Envoy {
+namespace ExtAuthz {
+namespace TcpFilter {
+
+InstanceStats Config::generateStats(const std::string& name, Stats::Scope& scope) {
+  std::string final_prefix = fmt::format("ext_authz.{}.", name);
+  return {ALL_TCP_EXT_AUTHZ_STATS(POOL_COUNTER_PREFIX(scope, final_prefix),
+                                  POOL_GAUGE_PREFIX(scope, final_prefix))};
+}
+
+void Instance::callCheck() {
+  envoy::service::auth::v2::CheckRequest request;
+  CreateCheckRequest::createTcpCheck(filter_callbacks_, request);
+
+  status_ = Status::Calling;
+  filter_callbacks_->connection().readDisable(true);
+  config_->stats().active_.inc();
+  config_->stats().total_.inc();
+
+  calling_check_ = true;
+  client_->check(*this, request, Tracing::NullSpan::instance());
```

Comment:
I think you might want to retain ownership of `request` until it has been sent, the gRPC client doesn't make a copy and only guarantees the send happens after you receive a response.